verifier/src/evidence.rs

```diff
@@ -635,7 +729,71 @@ mod test {
             - [ ] all of the following must be true:
               - [x] The TCB issuer chain was verified.
               - [x] The QE identity issuer chain was verified.
-              - [ ] The Quote issuer chain could not be verified: X509 certificate has expired"#;
+              - [ ] The Quote issuer chain could not be verified: X509 certificate has expired
+              - [x] The TCB info was verified for the provided key"#;
+        assert_eq!(format!("\n{displayable}"), textwrap::dedent(expected));
+    }
+
+    #[test]
+    fn evidence_verifier_fails_for_tcb_certificate_revoked() {
+        let time = "2023-06-08T15:55:15Z"
+            .parse::<DateTime>()
+            .expect("Failed to parse time");
+
+        let certificate_verifier = TestDoubleChainVerifier::fail_at_certificate("CN=Intel SGX TCB Signing,O=Intel Corporation,L=Santa Clara,STATEORPROVINCENAME=CA,C=US",
+         CertificateChainVerifierError::CertificateRevoked);
+
+        let verifier =
+            EvidenceVerifier::new(&certificate_verifier, [] as [TrustedIdentity; 0], time);
+
+        let quote_bytes = include_bytes!("../data/tests/hw_quote.dat");
+        let quote = Quote3::try_from(quote_bytes.to_vec()).expect("Failed to parse quote");
+        let collateral = collateral(TCB_INFO_JSON, QE_IDENTITY_JSON);
+        let evidence = Evidence::new(quote, collateral).expect("Failed to create evidence");
+
+        let verification = verifier.verify(&evidence);
+
+        assert_eq!(verification.is_success().unwrap_u8(), 0);
+
+        let displayable = VerificationTreeDisplay::new(&verifier, verification);
+        // Note that the TCB and QE identity happen to use the same certificate, so the QE identity
+        // will also fail.
+        let expected = r#"
+            - [ ] all of the following must be true:
+              - [ ] The TCB issuer chain could not be verified: X509 certificate has been revoked
+              - [ ] The QE identity issuer chain could not be verified: X509 certificate has been revoked
+              - [x] The Quote issuer chain was verified.
+              - [x] The TCB info was verified for the provided key"#;
```

Comment:
This test case exposed that returning the verifying key in the certificate chain verifier would result in
```
- [ ] all of the following must be true:
  - [ ] The TCB issuer chain could not be verified: X509 certificate has been revoked
  - [ ] The QE identity issuer chain could not be verified: X509 certificate has been revoked
  - [x] The Quote issuer chain was verified.
  - [ ] The TCB info could not be verified: Error verifying the signature
```